tfhe/src/integer/server_key/comparator.rs

```diff
@@ -21,6 +21,24 @@ enum MinMaxSelector {
     Min,
 }
 
+/// This function encodes how to reduce the ordering of two blocks.
+///
+/// It is used to generate the necessary lookup table.
+///
+/// `x` is actually two ordering values packed
+/// where in the first 2 lsb bits there is the ordering value of the less significant block,
+/// and the 2 msb bits of x contains the ordering of the more significant block.
+fn reduce_two_orderings_function(x: u64) -> u64 {
+    let msb = (x >> 2) & 3;
+    let lsb = x & 3;
```

Comment:
can we do better ? i.e. do we need two full bits for each packed ordering ?

From afar for an ordering you need lower, equal, greater or 3 values, which could be packed in a value mod 3 (0, 1, 2) and then if you add the other value scaled by 3 you can have the msb doing x / 3 and lsb with x % 3 ? means the max value we store would be 6 which fits in 3 bits ?

Reply:
We could, but that would not give room for improvements for 2_2 and we don't use parameters which have only 3 bits in integers, we jump from 2 to 4

Reply:
ok no worries